crowdfunding_compassion/controllers/projects_controller.py

```diff
@@ -7,31 +7,77 @@
 
 from ..forms.project_creation_form import NoGoalException,\
     NegativeGoalException, InvalidDateException, InvalidLinkException
+from odoo.addons.http_routing.models.ir_http import slug
+from odoo.addons.website.models.ir_http import sitemap_qs2dom
 
 
 class ProjectsController(Controller, FormControllerMixin):
-    @route("/projects", auth="public", website=True)
-    def get_projects_list(self, type=None, **kwargs):
-        project_obj = request.env["crowdfunding.project"]
+    _project_post_per_page = 12
 
-        # TODO connect pagination to backend -> CO-3213
-        return request.render(
-            "crowdfunding_compassion.project_list_page",
-            {"projects": project_obj.sudo().get_active_projects(type=type)},
-        )
+    def sitemap_projects(env, rule, qs):
+        projects = env['crowdfunding.project']
+        dom = sitemap_qs2dom(qs, '/projects', projects._rec_name)
+        dom += request.website.website_domain()
+        for f in projects.search(dom):
+            loc = '/project/%s' % slug(f)
+            if not qs or qs.lower() in loc:
+                yield {'loc': loc}
+
+    @route(["/projects", "/projects/page/<int:page>"], auth="public", website=True, sitemap=sitemap_projects)
+    def get_projects_list(self, type=None, page=1, year=None, **opt):
+        if request.website:
+            website_id = request.website.id
+            if website_id != 1:  # only for other site
+                domain = request.website.website_domain()
+                filters = list(filter(None, [
+                    ("state", "!=", "draft"),
+                    ("website_published", "=", True),
+                    ("deadline", ">=", datetime(year, 1, 1)) if year else None,
+                    ("deadline", "<=", datetime(year, 12, 31)) if year else None,
+                    ("type", "=", type) if type else None
+                ]))
+                filters += domain
+                project_obj = request.env["crowdfunding.project"]
+                total = project_obj.search_count(filters)
+
+                pager = request.website.pager(
+                    url='/projects',
+                    total=total,
+                    page=page,
+                    step=self._project_post_per_page,
+                    url_args={'type': type},
+                )
+                # TODO connect pagination to backend -> CO-3213
+                projects = project_obj.sudo()\
+                    .get_active_projects(type=type, domain=domain, offset=(page - 1) * self._project_post_per_page,
+                                             limit=self._project_post_per_page, page=page)
+                return request.render(
+                    "crowdfunding_compassion.project_list_page",
+                    {"projects": projects,
+
+                     "type": type,
+                     "pager": pager},
+                )
+            else:
+                raise werkzeug.exceptions.BadRequest()
```

Comment:
This way we will raise the error for any other situation where it didn't return.
```suggestion
        raise werkzeug.exceptions.BadRequest()
```